### ISSUE TITLE
`array-bracket-spacing` and array destructuring in function parameters with flow types

### ISSUE THREAD

**Issue by zaygraveyard** [babel/eslint-plugin-babel#35]
### test.js
```js
/*eslint array-bracket-spacing: [2, "never"] */
function fnArray([a, b]: Array<any >): void {
  // nothing important
}
```

### What versions I'm using
`eslint@1.10.3`
`babel-eslint@4.1.6`

### What I did
```sh
$eslint --parser babel-eslint test.js
```

### What I expected
No errors

### What actually happened
```
/tmp/test.js
  2:36  error  There should be no space before '>'  array-bracket-spacing

✖ 1 problem (1 error, 0 warnings)
```

### My Observations

The `ArrayPattern` node of the AST tree contains the type annotation.

```js
function fnArray([a, b]: Array<any >): void {
//               ^-----------------^
//                ArrayPattern
```

The `array-bracket-spacing` rule checks the last and before last tokens of the AST node (ie. last is `>` an before last is `any`).

Since the between the before last token and last token their is a space and the expected is no space, an error is reported.
